lib/rucio/core/did.py

```diff
@@ -3584,3 +3584,51 @@ def insert_deleted_dids(filter_, *, session: "Session"):
             is_archive=did.is_archive,
             constituent=did.constituent
         ).save(session=session, flush=False)
+        
+@read_session
+def container_to_datasets(
+    did_tuple: "Tuple[str, str]",
+    *,
+    session: "Session",
+    max_depth: int = 0,
+    _depth: int = 0,
```

Comment:
Indeed, placing max_depth and _depth after * was a simple error.
The motivation behind introducing max_depth and the _depth counter separately was to allow unlimited recursion. With only one parameter, you would have to set it to some very large number to make the recursion effectively unlimited, while with the current implementation you can set `max_depth` to <= 0 to tell the function to recurse indefinitely. That said, since there is the python limit of 1000 recursions anyway, there is no absolute need for two parameters, and I'm fine with reverting to your implementation.